src/petals/utils/linear8bitlt_patch.py

```diff
@@ -123,9 +125,156 @@ def custom_matmul8bitlt(
 
 
 class CustomMatMul8bitLt(MatMul8bitLt):
-    # forward is the same as in inference-only CxB
+    # forward is the same, but we added the fallback for pre-turing GPUs
     # backward is mostly the same, but adds one extra clause (see "elif state.CxB is not None")
 
+    @staticmethod
+    def forward(ctx, A, B, out=None, bias=None, state=CustomMatmulLtState):
```

Comment:
For simplicity: the new code in forward is limited to

- L133 - and a few times using_igemmlt was added to if-statemets
- L245-252

Reply:
For the reference, the original is [here](https://github.com/TimDettmers/bitsandbytes/blob/439f2b0c10abd3e9aade386d92810b074c69e9ec/bitsandbytes/autograd/_functions.py#L204).